### PR TITLE
ARCH context change to make compiling for multi platform

### PR DESCRIPTION
**Pull request by morlay**
[x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Now could compile go binaries for arm64 by

`TARGET_ARCH=arm64 BUILD_WITH_CONTAINER=1 DOCKERX_PUSH=1 make dockerx `

related: https://github.com/istio/istio/issues/21094#issuecomment-642363692
something need to updated for istio/proxy,  but this will be the first step.

### CODE REVIEW

**Comment by morlay** [comment]
@googlebot I signed it!

File: bin/init.sh [hunk 1]
```diff
@@ -142,10 +142,10 @@ ISTIO_ENVOY_RELEASE_URL=${ISTIO_ENVOY_RELEASE_URL:-${ISTIO_ENVOY_BASE_URL}/envoy
 ISTIO_ENVOY_LINUX_DEBUG_URL=${ISTIO_ENVOY_LINUX_DEBUG_URL:-${ISTIO_ENVOY_DEBUG_URL}}
 ISTIO_ENVOY_LINUX_RELEASE_URL=${ISTIO_ENVOY_LINUX_RELEASE_URL:-${ISTIO_ENVOY_RELEASE_URL}}
 # Variables for the extracted debug/release Envoy artifacts.
-ISTIO_ENVOY_LINUX_DEBUG_DIR=${ISTIO_ENVOY_LINUX_DEBUG_DIR:-"${OUT_DIR}/linux_amd64/debug"}
+ISTIO_ENVOY_LINUX_DEBUG_DIR=${ISTIO_ENVOY_LINUX_DEBUG_DIR:-"${OUT_DIR}/linux_${TARGET_ARCH}/debug"}
 ISTIO_ENVOY_LINUX_DEBUG_NAME=${ISTIO_ENVOY_LINUX_DEBUG_NAME:-"envoy-debug-${ISTIO_ENVOY_LINUX_VERSION}"}
 ISTIO_ENVOY_LINUX_DEBUG_PATH=${ISTIO_ENVOY_LINUX_DEBUG_PATH:-"${ISTIO_ENVOY_LINUX_DEBUG_DIR}/${ISTIO_ENVOY_LINUX_DEBUG_NAME}"}
-ISTIO_ENVOY_LINUX_RELEASE_DIR=${ISTIO_ENVOY_LINUX_RELEASE_DIR:-"${OUT_DIR}/linux_amd64/release"}
+ISTIO_ENVOY_LINUX_RELEASE_DIR=${ISTIO_ENVOY_LINUX_RELEASE_DIR:-"${OUT_DIR}/linux_${TARGET_ARCH}/${TARGET_ARCH}"}
```
Comment (howardjohn) [inline bin/init.sh:148]: shouldn't this be /release?

File: tools/buildx-gen.sh [hunk 2]
```diff
@@ -58,14 +58,17 @@ for file in "$@"; do
     cat <<EOF >> "${config}"
 target "$image-$variant" {
     context = "${out}/${file}"
+    
     dockerfile = "Dockerfile.$image"
-    tags = ["${HUB}/${image}:${tag}"]
+    tags = ["${HUB}/${image}:${tag}-${TARGET_ARCH}"]
```
Comment (howardjohn) [inline tools/buildx-gen.sh:63]: I don't think we want to change the image names for the standard arch, maybe just append it if a notstandard?

File: bin/init.sh (hunk 1 above)
Comment (morlay) [reply bin/init.sh:148]: yes, i will update this.

File: tools/buildx-gen.sh (hunk 2 above)
Comment (morlay) [reply tools/buildx-gen.sh:63]: we could change it by `tag=${version}-${arch}`. i will revert it.

cause the istio/build-tools images not support oci.
so multi stage docker build is not possible now, we have to build multi tag.

**Comment by morlay** [comment]
should define the tag for target arch

```
TARGET_ARCH=arm64 TAG=1.6.1-arm64 BUILD_WITH_CONTAINER=1 DOCKERX_PUSH=1 make dockerx
```

**Comment by morlay** [comment]
is that failed test run `common/scripts/setup_env.sh` ?

**Comment by morlay** [comment]
hope this could be merged. it will be make the arm64 ci automatedly.

* istio-envoy: https://github.com/querycap/istio-envoy-arm64
* istio-images: https://github.com/querycap/istio (now it is deps my fork of istio)

File: common/scripts/setup_env.sh [hunk 3]
```diff
@@ -73,10 +73,10 @@ TIMEZONE=$(readlink $readlink_flags /etc/localtime | sed -e 's/^.*zoneinfo\///')
 export TIMEZONE
 
 export TARGET_OUT="${TARGET_OUT:-$(pwd)/out/${TARGET_OS}_${TARGET_ARCH}}"
-export TARGET_OUT_LINUX="${TARGET_OUT_LINUX:-$(pwd)/out/linux_amd64}"
+export TARGET_OUT_LINUX="${TARGET_OUT_LINUX:-$(pwd)/out/linux_${TARGET_ARCH}}"
```
Comment (ericvn) [inline common/scripts/setup_env.sh:76]: Note that changes in this file are made in the common-files repo and are propagated to the other repos.

See https://github.com/istio/common-files/blob/master/files/common/scripts/setup_env.sh#L76 for example.

Comment (morlay) [reply common/scripts/setup_env.sh:76]: should i created i create another PR of that repo?
or just revert it

Comment (ericvn) [reply common/scripts/setup_env.sh:76]: You would create a PR on that repo with the change. Once that PR is merged, all the repos will be updated with the common file update. Then you can remove the file in this PR.

**Comment by tegk** [comment]
@morlay I would very much appreciate it if you could get this ready to be merged?

**Comment by morlay** [comment]
@tegk i hope this could be merged too. seems istio team have other things to do. 

i have to copy the main scripts for my custom build https://github.com/querycap/istio

**Pull request closed without merging**